### PR TITLE
save localStorage and mark visited link

### PR DESCRIPTION
**Pull request by bapboy18**

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -22,10 +22,10 @@
...
 //= require review
 
-$(document).ready(function(){
+$(document).ready(function() {
   $("#myModalName").html($("#new_address").data("name"));
   var directional = $("#new_address").data("directional");
-  switch(directional){
+  switch(directional) {
     case "new_address":
       getElementById("new_address_view").style.display="block";
```
Comment (ThuBM) [inline app/assets/javascripts/application.js]: display = "block"

File: app/assets/javascripts/application.js [hunk 2]
```diff
@@ -42,12 +42,12 @@ $(document).ready(function(){
 
       if(min_square === "" && max_square === "" &&
```
Comment (ThuBM) [inline app/assets/javascripts/application.js]: add space after if

File: app/assets/javascripts/application.js [hunk 3]
```diff
@@ -78,6 +78,75 @@ $(document).ready(function(){
...
+  var i = 0;
+
+  $(".visit").click(function() {
+    history_Storage($(this));
+  });
+
+  function history_Storage(visit) {
+    var adds_by_province = visit.data("adds-by-province");
+
+    if(localStorage.provinces != null) {
```
Comment (ThuBM) [inline app/assets/javascripts/application.js]: add space after if

File: app/assets/javascripts/application.js [hunk 4]
```diff
@@ -78,6 +78,75 @@ $(document).ready(function(){
...
+      provinces = JSON.parse(localStorage.provinces);
+      i = provinces.length;
+    }
+
+    if(not_in_array(adds_by_province)) {
```
Comment (ThuBM) [inline app/assets/javascripts/application.js]: add space after if

File: app/assets/javascripts/application.js [hunk 5]
```diff
@@ -78,6 +78,75 @@ $(document).ready(function(){
...
+      provinces[i] = adds_by_province;
+      localStorage["provinces"] = JSON.stringify(provinces);
+      i++;
+    }
+  }
+
+  function not_in_array(adds_by_province) {
+    for (var j = 0; j < provinces.length; j++) {
```
Comment (ThuBM) [inline app/assets/javascripts/application.js]: use $.inArray(value, array)

File: app/assets/javascripts/application.js [hunk 6]
```diff
@@ -78,6 +78,75 @@ $(document).ready(function(){
...
+      province_address.html(provinces[j].address);
+
+      capacity = html.find("span.capacity");
+      capacity.html(provinces[j].capacity);
+
+      contact = html.find("span.contact");
+      contact.html(provinces[j].contact);
+
+      date = html.find("span.right");
+      date.html(d.getFullYear() + "-" + (d.getMonth() + 1) + "-" + d.getDate()
```
Comment (ThuBM) [inline app/assets/javascripts/application.js]: sử dụng javascript date format

File: app/views/addresses/_adds_by_province.html.erb [hunk 7]
```diff
@@ -12,7 +8,9 @@
   <div class="address-details">
     <div class="title">
       <%= link_to adds_by_province.title, address_path(adds_by_province),
-        title: "adds_by_province.title" %>
+        title: "adds_by_province.title", class: "visit",
+        data: {adds_by_province: adds_by_province.as_json.merge({
+        image_tag: image_tag_link(adds_by_province)})} %>
```
Comment (ThuBM) [inline app/views/addresses/_adds_by_province.html.erb]: indent

File: app/assets/javascripts/application.js (hunk 4 above)
Comment (samnangcattor) [inline app/assets/javascripts/application.js]: I think  don't need to it

File: app/assets/javascripts/visited_news.js [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+  var i = 0;
+
+  $(".visit").click(function() {
+    save_link($(this));
+  });
+
+  function save_link(visit) {
+    var adds_by_province = visit.data("adds-by-province");
+
+    if (localStorage.provinces != null) {
```
Comment (dieunb) [inline app/assets/javascripts/visited_news.js]: `!==`

**Pull request merged**